main/tests/cypress/cypress/integration/project/grid/misc/expressions.spec.js

```diff
@@ -0,0 +1,306 @@
+		loadExpressionPanel()
+		cy.get('select[bind="expressionPreviewLanguageSelect"]').select(
+			'clojure'
+		)
+		typeExpression('(.. value (thisClojureFunctionDoesNotExists) )')
+		cy.get(
+			'.expression-preview-table-wrapper tr:nth-child(2) td:last-child'
+		).should('to.contain', 'Error: No matching method')
+	})
+
+	it('Test switching from one langage to another', function () {
+		cy.loadAndVisitProject('food.mini')
+		loadExpressionPanel()
+		typeExpression('(.. value (toLowerCase) )')
+		// error is expected, this is clojure language
+		cy.get('.expression-preview-parsing-status').should(
+			'to.contain',
+			'Parsing error'
+		)
+		// switching to clojure
+		cy.get('select[bind="expressionPreviewLanguageSelect"]').select(
+			'clojure'
+		)
+		cy.get('.expression-preview-parsing-status').should(
+			'not.to.contain',
+			'Parsing error'
+		)
+	})
+
+	it('Test the preview (GREL)', function () {
+		cy.loadAndVisitProject('food.mini')
+		loadExpressionPanel()
+		typeExpression('value.toLowercase()')
+
+		cy.get(
+			'.expression-preview-table-wrapper tr:nth-child(1) td:last-child'
+		).should('to.contain', 'value.toLowercase()')
+		cy.get(
+			'.expression-preview-table-wrapper tr:nth-child(2) td:last-child'
+		).should('to.contain', 'butter,with salt')
+		cy.get(
+			'.expression-preview-table-wrapper tr:nth-child(3) td:last-child'
+		).should('to.contain', 'butter,whipped,with salt')
+	})
+
+	it('Test the help tab', function () {
+		cy.loadAndVisitProject('food.mini')
+		loadExpressionPanel()
+		cy.get('#expression-preview-tabs li').contains('Help').click()
+		cy.get('#expression-preview-tabs-help').should('be.visible')
+		cy.get('#expression-preview-tabs-help').should(
+			'to.contain',
+			`The current cell. It has a few fields: 'value', 'recon' and 'errorMessage'.`
+		)
+	})
+
+	it('Test the history behavior, ensure expressions are stored', function () {
+		cy.loadAndVisitProject('food.mini')
+		// Because history is shared across projects, we need to use an expression that is unique
+
+		// Use a first unique expression
+		const uniqueExpression = 'value.testing_One_' + Date.now() + '()'
+		loadExpressionPanel()
+		typeExpression(uniqueExpression)
+		cy.get('.dialog-footer button').contains('OK').click()
+		// ensure the function has been added to the facet
+		cy.get('#refine-tabs-facets').contains(
+			uniqueExpression.replace('()', '')
+		)
+
+		// Reload and review history
+		// Ensure the previously used expression is listed
+		loadExpressionPanel()
+		cy.get('#expression-preview-tabs li').contains('History').click()
+		cy.get('#expression-preview-tabs-history')
+			.should('be.visible')
+			.should('to.contain', uniqueExpression)
+	})
+
+	it('Test the reuse of expressions from the history', function () {
+		cy.loadAndVisitProject('food.mini')
+		// Because history is shared across projects, we need to build and use an expression that is unique
+		const uniqueExpression = 'value.testing_One_' + Date.now() + '()'
```

Comment:
```suggestion
		const uniqueExpression = 'value + now() + "HistoryReusedSuccess"'
```

Reply:
Following your suggestion @thadguidry, I've changed the code to use only valid GREL expressions for history and star testing